pkgs/applications/virtualization/docker/compose.nix

```diff
@@ -20,6 +27,11 @@ buildGoModule rec {
     rm -rf pkg/e2e/
   '';
 
+  preBuild = lib.optionalString (extraPlugins != [ ]) ''
+    substituteInPlace vendor/github.com/docker/cli/cli-plugins/manager/manager_unix.go \
+      --replace-fail /usr/libexec/docker/cli-plugins "${pluginsRef}/libexec/docker/cli-plugins"
```

Comment:
Can we patch the code instead to fully rely on PATH or another env? Or do we maybe have such an env already in the code?